src/cmt_summary.c

```diff
@@ -130,7 +130,7 @@ double cmt_summary_quantile_get_value(struct cmt_metric *metric, int quantile_id
 {
     uint64_t val;
 
-    if (quantile_id < 0 || quantile_id > 5) {
+    if (quantile_id < 0 /*|| quantile_id > metric->sum_quantiles_count*/) {
```

Comment:
Is it possible the invalid read is because of this commented check? Even though valgrind reports this as happening on `cmt_summary_set_default`, it is possible the compiler inlined `cmt_summary_quantile_get_value`/ `cmt_summary_quantile_set_value`

Reply:
it is entirely possible. That check was commented while I worked out the best way to check the limit. The limit before was just hardcoded.